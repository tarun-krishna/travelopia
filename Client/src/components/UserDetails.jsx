import React from "react";
import axios from "axios";
import { useState, useEffect } from "react";
import "../styles/user.css";
const UserDetails = () => {
  // UserDetails Component
  const [list, setList] = useState([]);
  useEffect(() => {
    //useEffect to perform side effects to the website
    axios({
      //  Axios to fetch the data from mangodb database
      url: "/users",
    })
      .then((res) => {
        // handling promise
        setList(res.data.user);
      })
      .catch((err) => {
        console.log(err);
      });
  }, []);

  return (
    <div className="main">
      <div className="user-container">
        {list.map((item, _id) => {
          // map method to iterate throuh the list array
          return (
            <div key={item._id} className="user-cell">
              <div>
                <div>
                  <label className="detail-label">Name: {item.name}</label>
                </div>
                <div>
                  <label className="detail-label">E-mail: {item.email}</label>
                </div>
                <div>
                  <label className="detail-label">Place: {item.country}</label>
                </div>
              </div>
              <div>
                <div>
                  <label className="detail-label">
                    Travellers: {item.travellers}
                  </label>
                </div>
                <div>
                  <label className="detail-label">Budget: {item.budget}$</label>
                </div>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default UserDetails;
