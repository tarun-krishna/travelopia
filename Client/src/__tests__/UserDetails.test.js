import renderer from "react-test-renderer";
import UserDetails from "../components/UserDetails";

it("renders correctly", () => {
  const header1 = renderer.create(<UserDetails />).toJSON();
  expect(header1).toMatchSnapshot();
});
