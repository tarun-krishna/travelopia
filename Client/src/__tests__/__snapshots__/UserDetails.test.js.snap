// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="main"
>
  <div
    className="user-container"
  />
</div>
`;
